pkg/virt-controller/watch/vm_test.go

```diff
@@ -183,6 +183,11 @@ var _ = Describe("VirtualMachine", func() {
 			virtClient.EXPECT().AuthorizationV1().Return(k8sClient.AuthorizationV1()).AnyTimes()
 		})
 
+		shouldExpectGracePeriodPatched := func(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) {
+			patch := fmt.Sprintf(`{"spec":{"terminationGracePeriodSeconds": %d }}`, *vm.DeletionGracePeriodSeconds)
+			vmiInterface.EXPECT().Patch(context.Background(), vmi.Name, types.MergePatchType, []byte(patch), &metav1.PatchOptions{}).Return(vmi, nil)
+		}
+
```

Comment:
Sorry for the late review regarding this, but I think it is better to not pass the vm object.
```suggestion
		shouldExpectGracePeriodPatched := func(vmi *virtv1.VirtualMachineInstance, expactedGracePeriod int64) {
			patch := fmt.Sprintf(`{"spec":{"terminationGracePeriodSeconds": %d }}`, expactedGracePeriod)
			vmiInterface.EXPECT().Patch(context.Background(), vmi.Name, types.MergePatchType, []byte(patch), &metav1.PatchOptions{}).Return(vmi, nil)
		}

```

In this case the function call can be
```
shouldExpectGracePeriodPatched(vmi, v1.DefaultGracePeriodSeconds)
```

and  the changes at line 2043-2044 and  3303-3304 can be avoided.
With this, we are sure that we are using the default value, and not the explicit set one.
WDYT?

Reply:
I can't remove the changes at 2043-2044 and 3303-3304 because otherwise there is no value set for the DeletionGracePeriodSeconds and the test panics. I do agree though that we don't need to pass the whole VM object to the function, just the grace period value should be sufficient.